tools/cache/reflector.go

```diff
@@ -856,6 +856,10 @@ loop:
 			default:
 				utilruntime.HandleError(fmt.Errorf("%s: unable to understand watch event %#v", name, event))
 			}
+			if err != nil {
```

Comment:
this `if` block is a duplicate of line 827... the `err` variables in all the cases are local and don't affect the `err` variable inspected here... did you mean to return in the `default` case on line 857, and/or handle non-nil errs in all the event type cases?

(also, PRs are not accepted to this repo, you would need to open this against https://github.com/kubernetes/kubernetes/)

/close